src/Stratis.Bitcoin.Features.PoA/FederationManager.cs

```diff
@@ -19,12 +19,15 @@ public class FederationManager
 
         private readonly PoANetwork network;
 
+        private readonly PoAMiner miner;
+
         private readonly ILogger logger;
 
-        public FederationManager(NodeSettings nodeSettings, Network network, ILoggerFactory loggerFactory)
+        public FederationManager(NodeSettings nodeSettings, Network network, PoAMiner miner, ILoggerFactory loggerFactory)
         {
             this.settings = nodeSettings;
             this.network = network as PoANetwork;
+            this.miner = miner;
```

Comment:
Please make a note to add Guards here later.

Reply:
Kk. Adding guards in very next PR